extensions/applicationinsights-analytics-js/src/JavaScriptSDK/Telemetry/PageViewManager.ts

```diff
@@ -95,6 +96,24 @@ export class PageViewManager {
                     let location = getLocation();
                     uri = pageView.uri = location && location.href || "";
                 }
+
+                if (!firstPageViewSent){
+                    let perf = getPerformance();
+                    // Access the performance timing object
+                    const navigationEntries = (perf && perf.getEntriesByType && perf.getEntriesByType("navigation"));
+    
+                    // Edge Case the navigation Entries may return an empty array and the timeOrigin is not supported on IE
+                    if (navigationEntries && navigationEntries[0] && !isUndefined(perf.timeOrigin)) {
+                        // Get the value of loadEventStart
+                        const loadEventStart = (navigationEntries[0] as PerformanceNavigationTiming).loadEventStart;
+                        pageView.startTime =  new Date(perf.timeOrigin + loadEventStart);
+                    } else {
+                        // calculate the start time manually
+                        let duration = ((customProperties || pageView.properties || {}).duration || 0);
+                        pageView.startTime = new Date(new Date().getTime() - duration);
```

Comment:
This still needs to be done of ALL other cases, otherwise the reported "time" will be wrong.

Reply:
Otherwise the Portal will show the wrong "startTime"